### PR TITLE
xbox controller files via upload

### PR DESCRIPTION
**Pull request by rayjinghaolei**

### CODE REVIEW

File: joystick/launch/turtle_joy.launch [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<launch>
+	<!-- Turtlesim Node-->
+		<node pkg="turtlesim" type="turtlesim_node" name="sim"/>
```
Comment (wmmc88) [inline joystick/launch/turtle_joy.launch]: have your test packages outside of this package.

File: joystick/launch/turtle_joy.launch [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+	
+	<!-- joy node -->
+		<node respawn="true" pkg="joy"
+			type="joy_node" name="turtle_joy">
```
Comment (wmmc88) [inline joystick/launch/turtle_joy.launch]: don't call this turtle_joy. maybe `xbox_joy`

File: joystick/launch/turtle_joy.launch [hunk 3]
```diff
@@ -0,0 +1,18 @@
+<launch>
+	<!-- Turtlesim Node-->
+		<node pkg="turtlesim" type="turtlesim_node" name="sim"/>
+	
+	<!-- joy node -->
+		<node respawn="true" pkg="joy"
```
Comment (wmmc88) [inline joystick/launch/turtle_joy.launch]: this should all be in a group ns. maybe "/uwrt_basestation/"

File: joystick/src/general_templete.cpp [hunk 4]
```diff
@@ -0,0 +1,100 @@
+#include <ros/ros.h>
```
Comment (wmmc88) [inline joystick/src/general_templete.cpp]: what is this file? what does `general_templete` mean?

File: joystick/src/Controller.cpp [hunk 5]
```diff
@@ -0,0 +1,137 @@
+#include <ros/ros.h>
+#include <geometry_msgs/Twist.h>
+#include <sensor_msgs/Joy.h>
+
+class TeleopTurtle{
```
Comment (wmmc88) [inline joystick/src/Controller.cpp]: class definitions should be in a header file

File: joystick/src/Controller.cpp [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+
+		arm_end_eff_twist.linearx = joy->axes[cross_key_up_down_];
+		arm_end_eff_twist.lineary = joy->axes[cross_key_left_right_];
+		arm_end_eff_twist.linearz = joy->axes[cross_key_up_down_] == -1;
+		arm_publisher.pubish(arm_end_eff_twst)
+
+}
+
+int main(int argc, char** argv){
+	ros::init(argc, argv, "teleop_turtle");
```
Comment (wmmc88) [inline joystick/src/Controller.cpp]: why is the node name "teleop_turtle"?

File: joystick/src/Controller.cpp (hunk 5 above)
Comment (wmmc88) [inline joystick/src/Controller.cpp]: why is this class teleop turtle?

File: joystick/src/general_templete.cpp [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+		nh_.param("right_stick_horizontal_scale", right_stcik_left_right_scale, right_stcik_left_right_scale);
+		
+		vel_pub_ = nh_.advertise<geometry_msgs::Twist>("turtle1/cmd_vel", 1);
+		
+		joy_sub_ = nh_.subscribe<sensor_msgs::Joy>("joy", 10, &TeleopTurtle::joyCallback, this);
+		
+}
+
+	void TeleopTurtle::joyCallback(const sensor_msgs::Joy::ConstPtr& joy){
+	geometry_msgs::Twist twist;
```
Comment (wmmc88) [inline joystick/src/general_templete.cpp]: where's the handling for all the other buttons?

File: joystick/src/general_templete.cpp [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+	void joyCallback(const sensor_msgs::Joy::ConstPtr& joy);
+	ros::NodeHandle nh_;
+	
+	//axes
+	int left_stick_up_down, left_stick_left_right
+	LT,right_stick_left_right, right_stick_up_down, RT,
+	cross_key_left_right, cross_key_up_down;
+	
+	//buttons
+	int A, B, X, Y, LB, RB, start, back, power, button_stick_left, button_stick_right;
```
Comment (wmmc88) [inline joystick/src/general_templete.cpp]: `a_button_, b_button_, x_button_, y_button_, left_bumper_button_, right_bumper_button_, start_button_, back_button_, power_button_, left_stick_button_`

File: joystick/src/general_templete.cpp [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+class TeleopTurtle{
+	public:
+	TeleopTurtle();
+	
+	private:
+	void joyCallback(const sensor_msgs::Joy::ConstPtr& joy);
+	ros::NodeHandle nh_;
+	
+	//axes
+	int left_stick_up_down, left_stick_left_right
```
Comment (wmmc88) [inline joystick/src/general_templete.cpp]: not descriptive names. fix like suggestions below

File: joystick/src/turtle_teleop_joy.cpp [hunk 10]
```diff
@@ -0,0 +1,49 @@
+#include <ros/ros.h>
```
Comment (wmmc88) [inline joystick/src/turtle_teleop_joy.cpp]: what is the point of this file here?

File: joystick/src/Controller.cpp [hunk 11]
```diff
@@ -0,0 +1,137 @@
...
+		RB_(5),
+		start_(6),
+		back_(7),
+		power_(8),
+		button_stick_left_(9),
+		button_stick_right_(10)
+		
+	{
+		//axes
+		nh_.param("left_stick_vertical", left_stick_up_down_, left_stick_up_down_);
```
Comment (wmmc88) [inline joystick/src/Controller.cpp]: What are all these parameters being used for?

**Comment by rayjinghaolei** [comment]
> We should call tomorrow to discuss changes. I think you should probably have an interface object that u implement in different ways for each of the available layouts. and the joy cb should choose which of the subclasses objects to call.

Sure, when do you want to call?

**Pull request closed without merging**